### ISSUE TITLE
Move fastify-secure-session dependency to devDependencies

### ISSUE THREAD

**Issue by mgcrea** [fastify/fastify-passport#91]
## 🚀 Feature Proposal

Currently `fastify-passport` has `fastify-secure-session` as a dependency.

However, the package is only ever used to import its types or for testing.

Would be great to move the dep to devDependencies.

## Motivation

You can use `fastify-passport` with another session provider than `fastify-secure-session` (as long as you keep a compatible session API), in that case, `fastify-secure-session` won't ever be used. This can lead to issues if you have to target systems where building native modules is not an option (eg. windows devices where you don't fully control the system) since it depends on `sodium-native`.

## Example

On most of my projects, I need a Redis storage + signed cookie so I went with a custom solution fitting my needs:

- [@mgcrea/fastify-session](https://github.com/mgcrea/fastify-session)
- [@mgcrea/fastify-session-redis-store](https://github.com/mgcrea/fastify-session-redis-store)
- [@mgcrea/fastify-session-sodium-crypto](https://github.com/mgcrea/fastify-session-sodium-crypto)

On another one that must target windows-based devices, I'm using only

- [@mgcrea/fastify-session](https://github.com/mgcrea/fastify-session) that defaults to a memory-store + hmac and does not need compilation.

However I encountered the libsodium build issue and discovered it was a hard dependency of this project.

Will gladly send a PR if you consider my request.

**Comment by airhorns** [comment]
I think this is a great idea -- my intention was to flesh out the SessionManagers inside `fastify-passport` to explicitly support other session providers as opposed to just assuming they would all support the exact same API. I think it'd be awesome to add explicit support for other session providers in this repo, and I would also accept PRs that let you send in a custom session manager to use whatever you wanted to store the session. I think this library probably shouldn't couple to things outside the Fastify github org, so to use your session doodad there you'd send in a custom SessionManager that then used it underneath. 

Also, FWIW, I think it'd be great to get a good productionized server-side session store using Redis (or `abstract-cache`) that the community standardized on!

**Comment by mgcrea** [comment]
Thanks, I've just sent a PR #92 with only the basic dep -> devDep switch, would be awesome to have it tagged/release soon.

In my case I've designed the lib with the constraint of being hot-swappable with fastify-secure-session, so the current `SecureSessionManager` works for me. But indeed it would be great to promote other session managers, maybe we should start by adding a list of compatible session managers to the readme, and state that the project is open to contributions.

Supporting a custom `SessionManager` would be nice and even more flexible, I've looked at the code and it looks like it would need a bit of refactoring. I've started something in #93, we can continue the discussion here.

Regarding the redis store, would indeed be nice. I had fun writing it but not sure it is really production ready. I did see that there was an existing and popular [fastify-session](https://github.com/SerayaEryn/fastify-session) but I really wanted to have flexible options for cookie signature (play with libsodium that looks like to be significantly faster than traditional hmac based signing according to my tests), and it's a bit hard for me to switch back working too much on vanilla js after using typescript daily!

Anyway, thanks for the great lib, it convinced me to try fastify for my latest projects as it felt safer to have the auth part covered with something known after a decade of using express!
